GoogleForCreators/web-stories-wp
# Page Layouts: Toggle layout assets

## Context

Want to allow users to apply page layouts _with_ and _without_ the images included in the page layouts.

[Designs:](https://www.figma.com/file/NHHMwIfxMnz39NxqkrFb7R/Stories-Post-Stable?node-id=5101%3A456645)
<img src="https://user-images.githubusercontent.com/22185279/113062421-ef2cfe00-9170-11eb-9f1e-7c452c6980d0.png" width="400px" />

## Summary

- Adds new UI as show in designs above
- Clicking the toggle will add/remove image content from layouts
- Clicking the page layout will apply the layout with or without the image content

Fixed an accessibility concern in the text panes tab. Can be moved to another pr if we want it separate.

## Relevant Technical Choices

n/a

## To-do

none

## User-facing changes

|Before|After|
|--|--|
|![before](https://user-images.githubusercontent.com/22185279/113062756-7b3f2580-9171-11eb-9928-1a5acfade5c8.gif)|![after](https://user-images.githubusercontent.com/22185279/113062763-7e3a1600-9171-11eb-8822-f24cd83024a1.gif)|
|![before-rtl](https://user-images.githubusercontent.com/22185279/113063700-f7863880-9172-11eb-8f35-49332e4b4866.gif)|![after-rtl](https://user-images.githubusercontent.com/22185279/113063616-d9b8d380-9172-11eb-90fd-f69bdf73e432.gif)|


## Testing Instructions

1. Create a new story.
2. Click on the page layouts tab.
3. Click on a layout. Should add layout without the image content.
4. Clicking the toggle and the toggle label should change the toggle value.  Pressing `spacebar` when the toggle is focused should work too. (Note: there's some lag here as we re-create all the page layouts)
5. Click a page layout. Should add layout without the image content.

Things that should still work:
- changing a page layout when there is something on the canvas should show a dialog prompting you to accept deleting any existing stuff
- animations should still be preserved when the layout is applied
- Panel should still be keyboard accessible

### QA






This PR can be tested by following these steps:


1. Create a new story.
2. Click on the page layouts tab.
3. Click on a layout. Should add layout without the image content.
4. Clicking the toggle and the toggle label should change the toggle value.  Pressing `spacebar` when the toggle is focused should work too. (Note: there's some lag here as we re-create all the page layouts)
5. Click a page layout. Should add layout without the image content.

Things that should still work:
- changing a page layout when there is something on the canvas should show a dialog prompting you to accept deleting any existing stuff
- animations should still be preserved when the layout is applied
- Panel should still be keyboard accessible

### UAT






## Reviews

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist




---



Fixes #6130
